homeassistant/components/recorder/websocket_api.py

```diff
@@ -327,26 +322,17 @@ async def ws_adjust_sum_statistics(
     def valid_units(statistics_unit: str | None, display_unit: str | None) -> bool:
         if statistics_unit == display_unit:
             return True
-        if (
-            statistics_unit == DistanceConverter.NORMALIZED_UNIT
-            and display_unit in DistanceConverter.VALID_UNITS
-        ):
-            return True
-        if statistics_unit == ENERGY_KILO_WATT_HOUR and display_unit in (
-            ENERGY_MEGA_WATT_HOUR,
-            ENERGY_WATT_HOUR,
-        ):
-            return True
-        if (
-            statistics_unit == MassConverter.NORMALIZED_UNIT
-            and display_unit in MassConverter.VALID_UNITS
-        ):
-            return True
-        if (
-            statistics_unit == VolumeConverter.NORMALIZED_UNIT
-            and display_unit in VolumeConverter.VALID_UNITS
+        for converter in (
+            DistanceConverter,
+            EnergyConverter,
+            MassConverter,
+            VolumeConverter,
```

Comment:
I think we can just loop over `statistics.STATISTIC_UNIT_TO_UNIT_CONVERTER.values()` since that's the limitation imposed by `statistics.adjust_statistics`.